client/app/bundles/course/survey/pages/ResponseIndex/__test__/index.test.jsx

```diff
@@ -59,43 +55,39 @@ const responsesData = {
   survey: {
     id: 2,
     title: 'Test Responses Page',
+    start_at: '2017-03-01T09:10:01.180+08:00',
     end_at: '2017-03-02T09:10:01.180+08:00',
   },
 };
 
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useParams: () => ({
+    surveyId: responsesData.survey.id.toString(),
+    courseId: global.courseId.toString(),
+  }),
+}));
+
 beforeEach(() => {
   mock.reset();
 });
 
-const InjectedResponseIndex = connect((state) => ({
-  survey: state.surveys[0] || {},
-}))(ResponseIndex);
-
 describe('<ResponseIndex />', () => {
   it('allows responses to be saved', async () => {
     const surveyId = responsesData.survey.id;
-    const responseUrl = `/courses/${courseId}/surveys/${surveyId}/responses`;
-    mock.onGet(responseUrl).reply(200, responsesData);
-    const spyIndex = jest.spyOn(CourseAPI.survey.responses, 'index');
+    const url = `/courses/${global.courseId}/surveys/${surveyId}/responses`;
+    mock.onGet(url).reply(200, responsesData);
+    const spy = jest.spyOn(CourseAPI.survey.responses, 'index');
+
+    window.history.pushState({}, '', url);
+    const page = render(<ResponseIndex />);
+
+    await waitFor(() => expect(spy).toHaveBeenCalled());
 
-    // Mount response index page and wait for data to load
-    window.history.pushState({}, '', responseUrl);
-    const responseIndex = mount(
-      <MemoryRouter>
-        <InjectedResponseIndex />
-      </MemoryRouter>,
-      buildContextOptions(storeCreator({})),
-    );
-    await sleep(1);
+    responsesData.responses.forEach((response) => {
+      expect(page.getByText(response.course_user.name)).toBeVisible();
+    });
 
-    expect(spyIndex).toHaveBeenCalled();
-    responseIndex.update();
-    const tableBodies = responseIndex.find('ForwardRef(TableBody)');
-    const phantomStudentRows = tableBodies.at(2).find('ForwardRef(TableRow)');
-    const realStudentRows = tableBodies.at(1).find('ForwardRef(TableRow)');
-    const getStatus = (row) => row.find('td').at(1).text();
-    expect(getStatus(phantomStudentRows.first())).toBe('Submitted');
```

Comment:
Any reason why these checks were removed?

Reply:
Added back expectations for the response status.